src/faction.cpp

```diff
@@ -432,7 +432,7 @@ int npc::faction_display( const catacurses::window &fac_w, const int width ) con
     }
     std::string direction = direction_name( direction_from( player_abspos, guy_abspos ) );
     std::string centerstring = "center";
-    if( ( !direction.compare( centerstring ) ) == 0 ) {
+    if( direction != centerstring ) {
```

Comment:
You might as well use the string literal directly. There is no point in creating a string object and use it only once.